integration-tests/test_s3.py

```diff
@@ -7,19 +7,41 @@
 #
 
 from __future__ import unicode_literals
+import contextlib
 import io
 import os
 import subprocess
+import uuid
 
 import smart_open
 
-_S3_URL = os.environ.get('SO_S3_URL')
-assert _S3_URL is not None, 'please set the SO_S3_URL environment variable'
+_S3_BUCKET_NAME = os.environ.get('SO_BUCKET_NAME')
+assert _S3_BUCKET_NAME is not None, 'please set the SO_BUCKET_NAME environment variable'
 
+_SO_KEY = os.environ.get('SO_KEY')
+assert _SO_KEY is not None, 'please set the SO_KEY environment variable'
 
-def initialize_bucket():
+_S3_URL = 's3://%s/%s' % (_S3_BUCKET_NAME, _SO_KEY)
+
+
+@contextlib.contextmanager
+def temporary():
+    """Yields a bucket/key than can be used for temporary writing."""
+    key = '%s/%s' % (_SO_KEY, uuid.uuid4().hex)
+    url = 's3://%s/%s' % (_S3_BUCKET_NAME, key)
+    yield (_S3_BUCKET_NAME, url)
     subprocess.check_call(['aws', 's3', 'rm', '--recursive', _S3_URL])
 
+def case_test(function):
```

Comment:
`case_test` is counter-intuitive.

```suggestion
def test_case(function):
```